stl/inc/queue

```diff
@@ -108,18 +141,12 @@ public:
         _Swap_adl(c, _Right.c);
     }
 
-    template <class _Ty2, class _Container2>
-    friend bool operator==(const queue<_Ty2, _Container2>&, const queue<_Ty2, _Container2>&);
-    template <class _Ty2, class _Container2>
-    friend bool operator!=(const queue<_Ty2, _Container2>&, const queue<_Ty2, _Container2>&);
-    template <class _Ty2, class _Container2>
-    friend bool operator<(const queue<_Ty2, _Container2>&, const queue<_Ty2, _Container2>&);
-    template <class _Ty2, class _Container2>
-    friend bool operator>(const queue<_Ty2, _Container2>&, const queue<_Ty2, _Container2>&);
-    template <class _Ty2, class _Container2>
-    friend bool operator<=(const queue<_Ty2, _Container2>&, const queue<_Ty2, _Container2>&);
-    template <class _Ty2, class _Container2>
-    friend bool operator>=(const queue<_Ty2, _Container2>&, const queue<_Ty2, _Container2>&);
+    friend bool operator==<>(const queue&, const queue&);
+    friend bool operator!=<>(const queue&, const queue&);
+    friend bool operator</**/<>(const queue&, const queue&);
+    friend bool operator><>(const queue&, const queue&);
+    friend bool operator<=<>(const queue&, const queue&);
+    friend bool operator>=<>(const queue&, const queue&);

```

Comment:
Could we add spaces here instead? I had extreme difficulty mentally parsing this at first. Ditto `stack`.
```suggestion
    friend bool operator== <>(const queue&, const queue&);
    friend bool operator!= <>(const queue&, const queue&);
    friend bool operator< <>(const queue&, const queue&);
    friend bool operator> <>(const queue&, const queue&);
    friend bool operator<= <>(const queue&, const queue&);
    friend bool operator>= <>(const queue&, const queue&);
```

Reply:
I can add them, but clang-format will remove them again. How about:
```suggestion
    // clang-format off
    friend bool operator== <>(const queue&, const queue&);
    friend bool operator!= <>(const queue&, const queue&);
    friend bool operator<  <>(const queue&, const queue&);
    friend bool operator>  <>(const queue&, const queue&);
    friend bool operator<= <>(const queue&, const queue&);
    friend bool operator>= <>(const queue&, const queue&);
    // clang-format on
```

Reply:
I think that's acceptable justification for a clang-format off. However, I would recommend filing a clang-format bug if it's trying to smash `operator< <>` into `operator<<>` which (I believe) is a destructive transformation, not addressed by C++11 angle bracket disambiguation.

Reply:
Filed LLVM-45256. I nevertheless don't plan to mark this as `TRANSITION`, since I think the extra spaces in these declarations make them a bit more readable. (I'm perfectly happy with e.g. `operator>=<>(`, but I imagine most folks would not be.) Shout if you disagree.